packages/liveblocks-react/src/room.tsx

```diff
@@ -996,136 +996,118 @@ export function createRoomContext<
+                room.id,
+                settings,
+                notificationSettingsQuery
+              );
+            })
+            .catch(() => {
+              // TODO: Handle error
+            })
+        );
+      }
 
-    poller.start(POLLING_INTERVAL);
-  }
+      const lastRequestedAt = room[kInternal].comments.lastRequestedAt;
+      if (lastRequestedAt === null) return;
+
+      // Retrieve threads that have been updated/deleted since the last requestedAt value
+      requests.push(
+        room
+          .getThreads({ since: lastRequestedAt })
+          .then((result) => {
+            store.updateThreadsAndNotifications(
+              result.threads,
+              result.inboxNotifications,
+              result.deletedThreads,
+              result.deletedInboxNotifications
+            );
 
-  function decrementQuerySubscribers(queryKey: string) {
-    const requestCache = requestsCache.get(queryKey);
+            // If this query finished after the room was unmounted, we do not want to update the `lastRequestedAt` value
+            const room = client.getRoom(roomId);
+            if (room === null) return;
 
-    if (requestCache === undefined || requestCache.subscribers <= 0) {
-      console.warn(
-        `Internal unexpected behavior. Cannot decrease subscriber count for query "${queryKey}"`
+            room[kInternal].comments.lastRequestedAt = result.meta.requestedAt;
+          })
+          .catch(() => {
+            // TODO: Handle error
+          })
       );
-      return;
-    }
-
-    requestCache.subscribers--;
-
-    let totalSubscribers = 0;
-    for (const requestCache of requestsCache.values()) {
-      totalSubscribers += requestCache.subscribers;
-    }
+    });
 
-    if (totalSubscribers <= 0) {
-      poller.stop();
-    }
+    await Promise.allSettled(requests);
   }
 
-  async function getOrInitRequest<TRequestResult>(
+  async function getThreadsAndInboxNotifications(
+    roomId: string,
     queryKey: string,
-    requestFactory: () => Promise<TRequestResult>,
-    onSuccess: (requestResult: TRequestResult) => void
-  ): Promise<any> {
-    const requestInfo = requestsCache.get(queryKey);
-
-    if (requestInfo !== undefined) {
-      return requestInfo.promise;
-    }
+    options: UseThreadsOptions<TThreadMetadata>
+  ) {
+    const room = client.getRoom(roomId);
+    if (room === null) return;
 
-    const promise = requestFactory();
+    const queries = room[kInternal].comments.queries;
 
-    requestsCache.set(queryKey, {
-      promise,
-      requestFactory,
-      onSuccess,
-      subscribers: 0,
-    });
+    // If the query has already been recorded in the room, we do not make another fetch request
+    if (queries.has(queryKey)) return;
 
-    store.setQueryState(queryKey, {
-      isLoading: true,
-    });
+    // Add the query to the room's queries set to prevent duplicate fetch requests
+    queries.add(queryKey);
```

Comment:
We currently add a query key to the set of queries before the request starts. There is a possibility that the request might fail, in which case we want to maybe remove this `queryKey` from the set of queries. This could be a part of error handling logic - will have to create a separate PR to think about error handling in general.

Reply:
Same for `useInboxNotifications`